### PR TITLE
Retire TSV loading workflow

### LINKED ISSUE #548: Trigger TSV loading in ingestion workflow DAGs
## Problem

The various ingestion workflow DAGs (`europeana_ingestion_workflow`, `flickr_ingestion_workflow`, `wikimedia_ingestion_workflow`) are used to help update _old_ data for `dated` DAGs, which otherwise only consume data for the current date. They work by calculating a list of `reingestion_days`, generally weighted toward more recent data (so for example, we'll reingest every day in the past week, and then increasingly sparsely as we go further back in time), and then pulling data for those days.

The DAGs all run what amounts to just the `pull_data` task for each of these days, meaning a tsv is generated but not loaded into the upstream db. That happens completely separately in the `tsv_to_postgres_loader` DAG. This is undesirable for reasons described in #357 (which resolved the issue for our standard provider DAGs).

## Description

If we could move the tsv loading steps into the ingestion DAGs, we could get rid of the `tsv_to_postgres_loader` once and for all 😄 

A couple of options:
* We could use the [TriggerDagOperator](https://www.astronomer.io/guides/cross-dag-dependencies/) to simply trigger the corresponding provider dag (ie `europeana_ingestion_workflow` triggers `europeana_workflow`) for each of the `reingestion_days`.
  * If we go that route, we'll need to take a close look at the `max_active_tasks` for these DAGs. Currently most `workflow` DAGs have `max_active_tasks=1`, while the `ingestion` DAGs bump this up to allow for parallelization.
* Alternatively, we could keep the DAGs separate and refactor out the loading code so it is reusable. Then within the ingestion DAGs we could parallelize just the `pull_data` tasks and then run the `load_data` steps once for _all_ the consumed data
  * This will result in a much longer-running `load_data` task
  * Some refactoring may be required to ensure `load_data` can pull data from many upstream `pull_data` tasks.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #548 by @stacimc

## Description

This PR retires the `tsv_to_postgres_loader` DAG, since it is no longer used in any of our ingestion workflows :tada: I've also removed some no-obsolete code related to TSV pathing - this is covered by the DAG's XComs in the way our current pipelines are set up.

## Testing Instructions

1. `just test`
2. Run both regular and legacy provider DAG scripts and check that records still make their way into the database (I tried SMK, Brooklyn, and Jamendo)
3. Ensure that the `tsv_to_postgres_workflow` doesn't show up in your DAGs list locally.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by stacimc — APPROVED** [review]
🎉 

Tested Smithsonian and Stocksnap just to be thorough. Great!

File: DAGs.md [hunk 1]
```diff
@@ -536,56 +534,6 @@ Notes:                  https://stocksnap.io/api/load-photos/date/desc/1
                         API is undocumented.
 
 
-## `tsv_to_postgres_loader`
-
-#### Database Loader DAG
-**DB Loader Apache Airflow DAG** (directed acyclic graph) takes the media data saved
-locally in TSV files, cleans it using an intermediate database table, and saves
```
Comment (krysal) [inline DAGs.md:543]: I was about to ask if we don't need to conserve this documentation but I see we have it where it was moved :)
https://github.com/WordPress/openverse-catalog/blob/3217ed5ecf3f9babeb96929e8883110e5da1c199/openverse_catalog/dags/database/loader_workflow.py#L2-L6

**Review by krysal — APPROVED** [review]
LGTM! Tested the WordPress Photo Directory script and it loads rows in the image table 👍

File: DAGs.md (hunk 1 above)
Comment (AetherUnbound) [reply DAGs.md:543]: I made sure to check that as well 😄 Should have made a note in the description, apologies!

**Pull request merged**